### PR TITLE
Migrate ContentResolverUtilTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate ContentResolverUtilTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 1]
```diff
@@ -13,84 +13,71 @@
...
-
-        String filename = ContentResolverUtil.getFileName(mock, uri);
-
-        assertThat(filename, is("filename_from_cursor.jpg"));
+    fun testViaQueryWorking() {
+        val uri = Uri.parse("http://example.com/test.jpeg")
+        val mock = Mockito.mock(ContentResolver::class.java)
+        setQueryReturning(mock, cursorReturning("filename_from_cursor.jpg"))
+        val filename = getFileName(mock, uri)
+        MatcherAssert.assertThat(filename, Is.`is`("filename_from_cursor.jpg"))
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:44]: kotlincleanup is -> equalto

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 2]
```diff
@@ -13,84 +13,71 @@
...
-        Uri uri = mock(Uri.class);
-        when(uri.getScheme()).thenReturn(SCHEME_CONTENT);
-
-        ContentResolver mock = mock(ContentResolver.class);
-        setQueryThrowing(mock, new SQLiteException("no such column: _display_name (code 1 SQLITE_ERROR[1]): , " +
-                "while compiling: SELECT _display_name FROM ClipboardImageTable WHERE (id=855) ORDER BY _data"));
-
-        when(mock.getType(any())).thenReturn("image/gif");
+        val uri = Mockito.mock(Uri::class.java)
+        Mockito.`when`(uri.scheme).thenReturn(ContentResolver.SCHEME_CONTENT)
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:52]: kotlincleanup when -> whenever - should annotate the whole class with it, it is in a couple methods

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thank you!
Has a couple kotlincleanups to do post migration

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 3]
```diff
@@ -13,84 +13,74 @@
...
-
-        setQueryReturning(mock, cursorReturning("filename_from_cursor.jpg"));
-
-        String filename = ContentResolverUtil.getFileName(mock, uri);
-
-        assertThat(filename, is("filename_from_cursor.jpg"));
+    @KotlinCleanup("is --> equalTo")
+    fun testViaQueryWorking() {
+        val uri = Uri.parse("http://example.com/test.jpeg")
+        val mock = Mockito.mock(ContentResolver::class.java)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:43]: import: `Mockito`

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 4]
```diff
@@ -13,84 +13,74 @@
...
+        setQueryReturning(mock, cursorReturning("filename_from_cursor.jpg"))
+        val filename = getFileName(mock, uri)
+        MatcherAssert.assertThat(filename, Is.`is`("filename_from_cursor.jpg"))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:46]: import: `MatcherAssert`, `Is`

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 5]
```diff
@@ -13,84 +13,74 @@
...
+        Mockito.`when`(uri.scheme).thenReturn(ContentResolver.SCHEME_CONTENT)
+        val mock = Mockito.mock(ContentResolver::class.java)
+        setQueryThrowing(
+            mock,
+            SQLiteException(
+                "no such column: _display_name (code 1 SQLITE_ERROR[1]): , " +
+                    "while compiling: SELECT _display_name FROM ClipboardImageTable WHERE (id=855) ORDER BY _data"
+            )
+        )
+        Mockito.`when`(mock.getType(ArgumentMatchers.any())).thenReturn("image/gif")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:64]: import: `ArgumentMatchers`

**Review by david-allison — CHANGES_REQUESTED** [review]
The newlines in the Java have been removed in the conversion, could you add them back in

**Comment by oyeraghib** [comment]
@david-allison Android Studio is suggesting to convert the methods to private and make the class open (final). Should I do that ?

ss2

**Comment by david-allison** [comment]
> @david-allison Android Studio is suggesting to convert the methods to private and make the class open (final). Should I do that ?
> 
> ss2

Add `@KotlinCleanup("IDE-based lint")`

File: AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt [hunk 6]
```diff
@@ -13,84 +13,83 @@
...
-        String filename = ContentResolverUtil.getFileName(mock, uri);
+        val filename = getFileName(mock, uri)
 
-        assertThat(filename, is("filename_from_cursor.jpg"));
+        assertThat(filename, `is`("filename_from_cursor.jpg"))
     }
 
     @Test
-    public void testViaMimeType() {
+    @KotlinCleanup("is --> equalTo")
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/ContentResolverUtilTest.kt:56]: Feel free to stick this on the class in future

**Review by david-allison — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
thank you!

**Pull request merged**